### PR TITLE
feat(exercise-plugin): allow exercises in content

### PR DESCRIPTION
**Pull request by elbotho**
(based on https://github.com/serlo/frontend/pull/3052 and https://github.com/serlo/frontend/pull/3044, merge after) 

- toolbar for exercise plugin
- interactive-exercise pluging: move delete to toolbar
- move interactive chooser to exercise toolbar 

### demo

- [in preview](https://frontend-git-allow-exercises-everywhere-serlo.vercel.app/___editor_preview?name=%7B"plugin"%3A"rows"%2C"state"%3A%5B%7B"plugin"%3A"exercise"%2C"state"%3A%7B"content"%3A%7B"plugin"%3A"rows"%2C"state"%3A%5B%7B"plugin"%3A"text"%2C"state"%3A%5B%7B"type"%3A"p"%2C"children"%3A%5B%7B"text"%3A""%7D%5D%7D%5D%2C"id"%3A"6b4bca7c-1c57-4c87-944d-4d86b9817827"%7D%5D%2C"id"%3A"0070933c-8ca2-41a1-b2d0-e15ecdf38168"%7D%2C"interactive"%3A%7B"plugin"%3A"scMcExercise"%2C"state"%3A%7B"isSingleChoice"%3Afalse%2C"answers"%3A%5B%5D%7D%2C"id"%3A"1d19701e-798e-4ed6-8bec-0391d6ccb680"%7D%7D%2C"id"%3A"ee69d07b-ad4a-4798-8f6e-080af4acd8cc"%7D%5D%7D)
- [standalone exercise view](https://frontend-git-allow-exercises-everywhere-serlo.vercel.app/entity/create/Exercise/60730)
- [in normal editor](https://frontend-git-allow-exercises-everywhere-serlo.vercel.app/entity/repository/add-revision/277232)

### CODE REVIEW

File: src/serlo-editor/plugins/solution/editor.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { useEditorStrings } from '@/contexts/logged-in-data-context'
+import { selectIsDocumentEmpty, useAppSelector } from '@/serlo-editor/store'
+
+export function SolutionEditor({ editable, state, focused }: SolutionProps) {
+  const { prerequisite, strategy, licenseId } = state
+  const hasStrategy = !useAppSelector((state) =>
+    selectIsDocumentEmpty(state, strategy.id)
+  )
+
+  const isSerlo = true // TODO: what is a good way to only activate this when using from within serlo?
```
Comment (LarsTheGlidingSquirrel) [inline src/serlo-editor/plugins/solution/editor.tsx]: Maybe similar to `editorPlugins`? We could have a file `is-serlo.ts` and the value can be set to true in `SerloEditor` (or further up) once at the start. In the RLP-Integration we either set it to false or not set it at all.

File: src/serlo-editor/editor-ui/preview-overlay-simple.tsx [hunk 2]
```diff
@@ -4,19 +4,22 @@ import { EditableContext } from '../core'
 
 interface PreviewOverlaySimpleProps {
   active: boolean
+  fullOpacity: boolean
   children: React.ReactNode
 }
 
 export function PreviewOverlaySimple({
   active,
```
Comment (LarsTheGlidingSquirrel) [inline src/serlo-editor/editor-ui/preview-overlay-simple.tsx]: Was slightly confused to learn that when PreviewOverlaySimple is `active = true`, there is no overlay. But there is if `active = false`. Kind of flipped. Might help to rename it to just `PreviewSimple` or something similar.

File: src/serlo-editor/plugins/exercise/toolbar/toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  id,
+  state,
+  interactiveExerciseTypes,
+}: ExerciseProps & {
+  interactiveExerciseTypes: InteractiveExerciseType[]
+}) => {
+  const { interactive } = state
+  const exTemplateStrings = useEditorStrings().templatePlugins.exercise
+
+  const currentySelected = interactive.defined
```
Comment (LarsTheGlidingSquirrel) [inline src/serlo-editor/plugins/exercise/toolbar/toolbar.tsx]: Typo, should probably be `currentlySelected`

**Review by LarsTheGlidingSquirrel — APPROVED** [review]
I added a few comments. Other than that the code looks good to me!

**Review by Entkenntnis — APPROVED** [review]
Changes look good

**Pull request merged**